test/pylib/util.py

```diff
@@ -255,3 +259,22 @@ async def wait_for_first_completed(coros: list[Coroutine]):
         t.cancel()
     for t in done:
         await t
+
+
+def ninja(target):
+    """Build specified target using ninja"""
+    build_dir = 'build'
```

Comment:
Nope, it will not work. The problem here will be with the logs' directory. By default, it set to './testlog' and if there is no such directory it will fail. Since the current way of working mimicking the way of working of the test.py it works only from the root directory of the project.